crates/argmin/src/core/executor.rs

```diff
@@ -313,6 +313,7 @@ where
         mode: ObserverMode,
     ) -> Self {
         self.observers.push(observer, mode);
+        self.timer = true;
```

Comment:
Apologies, I probably miscommunicated this: I'd prefer it if timer was off by default, even if an observer is added. The only case where `timer` should be set to `true` by default is when a timeout is set.

Reply:
Should be fixed.